tests/tasks/test_notebook.py

```diff
@@ -150,25 +151,61 @@ def test_notebook_conversion_stores_as_unicode(tmp_directory, monkeypatch):
 
 
 @pytest.mark.parametrize(
-    "name, out_dir",
+    "name, out_dir, executor",
     [
-        ["sample.py", "."],
-        ["sample.R", "."],
-        ["sample.ipynb", "."],
+        ["sample.py", ".", "papermill"],
+        ["sample.R", ".", "papermill"],
+        ["sample.ipynb", ".", "papermill"],
         # check still works even if the folder does not exit yet
-        ["sample.ipynb", "missing_folder"],
+        ["sample.ipynb", "missing_folder", "papermill"],
+        # For Ploomber executor
+        ["sample.py", ".", "ploomber-engine"],
+        ["sample.R", ".", "ploomber-engine"],
+        ["sample.ipynb", ".", "ploomber-engine"],
+        # check still works even if the folder does not exit yet
+        ["sample.ipynb", "missing_folder", "ploomber-engine"],
+        ["sample.py", ".", "some-other-executor"],
+        ["sample.R", ".", "some-other-executor"],
+        ["sample.ipynb", ".", "some-other-executor"],
     ],
 )
-def test_execute_sample_nb(name, out_dir, tmp_sample_tasks):
+def test_execute_sample_nb(name, out_dir, executor, tmp_sample_tasks):
     dag = DAG()
 
-    NotebookRunner(
-        Path(name), product=File(Path(out_dir, name + ".out.ipynb")), dag=dag
-    )
-    dag.build()
+    nb_source = NotebookSource(Path(name))
+
+    if executor not in ["ploomber-engine", "papermill"]:
+        with pytest.raises(ValueError) as excinfo:
+            NotebookRunner(
+                Path(name),
+                product=File(Path(out_dir, name + ".out.ipynb")),
+                dag=dag,
+                executor=executor,
+            )
+            dag.build()
+        assert "Invalid executor" in str(excinfo.value)
+    elif nb_source.language == "r" and executor == "ploomber-engine":
+        with pytest.raises(Exception) as excinfo:
+            NotebookRunner(
+                Path(name),
+                product=File(Path(out_dir, name + ".out.ipynb")),
+                dag=dag,
+                executor=executor,
+            )
+            dag.build()
+
+        assert "NameError: name 'c' is not defined" in str(excinfo.value)
```

Comment:
looks like this error is coming from python but it isn't clear enough. Inside the notebook you can get the language and then raise a custom error like "ploomber engine is not supported for R notebooks, change the executor to papermill"

Reply:
I think this problem deserves a separate issue of its own

Reply:
why? it's within the scope of the PR. if somebody attempts to run an R notebook with ploomber-engine, they'll encounter a cryptic error.

Reply:
Sure, I would be happy to implement it. It will take me some time to understand the changes and flow, as the PR has been stale for over a week. 

Thanks for your patience